### PR TITLE
feat(linter): eslint/max-params

### PR DESCRIPTION
**Pull request by JoSeBu1**
Rule detail: https://eslint.org/docs/latest/rules/max-params

Maybe I can add function name, I was searching for it, but I don't know how to get it.

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/max_params.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                .and_then(Value::as_number)
+                .and_then(serde_json::Number::as_u64)
+                .map_or(3, |v| usize::try_from(v).unwrap_or(3));
+
+            Self(Box::new(MaxParamsConfig { max }))
+        }
+    }
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::Function(function) => {
```
Comment (Dunqing) [inline crates/oxc_linter/src/rules/eslint/max_params.rs:80]: You can get the function name by `function.id`

Comment (JoSeBu1) [reply crates/oxc_linter/src/rules/eslint/max_params.rs:80]: Mmm, ok, and for ArrowFunction? I think it doesn't have id property

Comment (mysteryven) [reply crates/oxc_linter/src/rules/eslint/max_params.rs:80]: Feels like we can ignore the name for `ArrowFunction`? [ESLint](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG1heC1wYXJhbXM6IFtcImVycm9yXCIsIDNdKi9cbi8qZXNsaW50LWVudiBlczYqL1xuXG5mdW5jdGlvbiBmb28xIChiYXIsIGJheiwgcXV4LCBxeHgpIHtcbiAgICBkb1NvbWV0aGluZygpO1xufVxuXG5sZXQgZm9vMiA9IChiYXIsIGJheiwgcXV4LCBxeHgpID0+IHtcbiAgICBkb1NvbWV0aGluZygpO1xufTsifQ==) also omits it.

Comment (JoSeBu1) [reply crates/oxc_linter/src/rules/eslint/max_params.rs:80]: wow, I didn't know this tool!, I have already applied the change

**Review by mysteryven — APPROVED** [review]
Thank you!

**Pull request merged**